### PR TITLE
Discover: Add more searchable keywords to resource select screen

### LINKED ISSUE #26706: Discover: Add more searchable keywords to resource select screen
Allow searching for keywords  `ssh`

### PR DESCRIPTION
**Pull request by kiosion**
## Changes
- Replace concatenated strings with arrays for Discover `ResourceSpec` `keyword` fields
- Add `ssh`, `osx`, `k8s`, & a few other variants to respective resources

Enterprise gravitational/teleport.e#4861

Resolves #26706

### CODE REVIEW

File: web/packages/teleport/src/Discover/SelectResource/databases.tsx [hunk 1]
```diff
@@ -26,11 +26,15 @@ import { ResourceKind } from '../Shared/ResourceKind';
...
 import { ResourceSpec, DatabaseLocation, DatabaseEngine } from './types';
 
-const baseDatabaseKeywords = 'db database databases';
-const awsKeywords = baseDatabaseKeywords + 'aws amazon web services';
-const gcpKeywords = baseDatabaseKeywords + 'gcp google cloud provider';
-const selfhostedKeywords = baseDatabaseKeywords + 'self hosted self-hosted';
-const azureKeywords = baseDatabaseKeywords + 'microsoft azure';
+const baseDatabaseKeywords = ['db', 'database', 'databases'];
+const awsKeywords = [...baseDatabaseKeywords, 'aws', 'amazon web services'];
+const gcpKeywords = [...baseDatabaseKeywords, 'gcp', 'google cloud provider'];
```
Comment (zmb3) [inline web/packages/teleport/src/Discover/SelectResource/databases.tsx:31]:
```suggestion
const gcpKeywords = [...baseDatabaseKeywords, 'gcp', 'google cloud platform'];
```

File: web/packages/teleport/src/Discover/SelectResource/databases.tsx [hunk 2]
```diff
@@ -93,7 +97,7 @@ export const DATABASES_UNGUIDED: ResourceSpec[] = [
   {
     dbMeta: { location: DatabaseLocation.Aws, engine: DatabaseEngine.DynamoDb },
     name: 'DynamoDB',
-    keywords: awsKeywords + 'dynamodb',
+    keywords: [...awsKeywords, 'dynamodb'],
```
Comment (zmb3) [inline web/packages/teleport/src/Discover/SelectResource/databases.tsx:100]: Will `dynamo` match `dynamodb`?

**Review by zmb3 — APPROVED** [review]

File: web/packages/teleport/src/Discover/SelectResource/databases.tsx (hunk 2 above)
Comment (kiosion) [reply web/packages/teleport/src/Discover/SelectResource/databases.tsx:100]: Yup

File: web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx [hunk 3]
```diff
@@ -82,14 +82,12 @@ export function SelectResource({ onSelect }: SelectResourceProps) {
...
-    const foundResources = list.filter(r => {
-      const match = split.every(s => r.keywords.includes(s));
-      if (match) {
-        return r;
-      }
-    });
-    setResources(foundResources);
+    const search = s.split(' ').map(s => s.toLowerCase());
+    const found = list.filter(r =>
+      search.every(s => r.keywords?.some?.(k => k.toLowerCase().includes(s)))
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx:87]: is checking for keywords necessary? we do say with types that it's required 🤔

**Review by kimlisa — APPROVED** [review]
just FYI since this will be your first time updating enterprise:

- typically merge OSS PR first, and then enterprise PR
- i make sure to have both the OSS and Enterprise PRs are both approved and ready to merge. this is because merging the OSS PR will sometimes break builds for other devs, so it's best to follow up with the enterprise PR as soon as possible
- After enterprise merges, update the e reference in the OSS repo. After making a PR, i go to `dev-teleport` [slack channel ](https://gravitational.slack.com/archives/C0DF0TPMY/p1723149904159469) and ask anyone to just approve this PR b/c faster the better

File: web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx (hunk 3 above)
Comment (kiosion) [reply web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx:87]: Shouldn't be... is there a downside to keeping this optional chaining in?

**Pull request merged**